### ISSUE TITLE
Every test run creates a new unique set of parallel databases

### ISSUE THREAD

**Issue by javan** [rails/rails#39291]
### Steps to reproduce
1. Create a new app using Rails master and mysql (may apply to others too). I tested with:
    ```ruby
    gem "rails", github: "rails/rails", ref: "459db0a69e1537ac484c26b252ff40543dc0aaed"
    ```
1. Run the tests, list the databases, repeat:
    ```sh
    $ rails test
    ...

    $ mysql -u root -e "show databases;" | grep blah
    blah_development
    blah_test
    blah_test-60fd9bfc-1b06-4de9-91a0-c7aca2232029
    blah_test-83a8b72f-9d4a-45a3-ba39-8a16ff576dea
    blah_test-9ad84ce2-b2cf-4774-9085-1d02256c0860
    blah_test-ed192204-5b14-4b85-b874-0accf893cee4
    ```

    ```sh
    $ rails test
    ...

    $ mysql -u root -e "show databases;" | grep blah
    blah_development
    blah_test
    blah_test-200dfea9-15af-48f9-9191-72403b740240
    blah_test-60fd9bfc-1b06-4de9-91a0-c7aca2232029
    blah_test-6737c1c9-905d-4bc0-90f8-86d234971ce1
    blah_test-6ea7d328-b90b-4840-aaa6-ed233d15fc55
    blah_test-83a8b72f-9d4a-45a3-ba39-8a16ff576dea
    blah_test-9ad84ce2-b2cf-4774-9085-1d02256c0860
    blah_test-cf265a0d-060c-4bf5-a80d-188ccc5fd13c
    blah_test-ed192204-5b14-4b85-b874-0accf893cee4
    ```

    ```sh
    $ rails test
    ...

    $ mysql -u root -e "show databases;" | grep blah
    blah_development
    blah_test
    blah_test-200dfea9-15af-48f9-9191-72403b740240
    blah_test-60fd9bfc-1b06-4de9-91a0-c7aca2232029
    blah_test-6737c1c9-905d-4bc0-90f8-86d234971ce1
    blah_test-6ea7d328-b90b-4840-aaa6-ed233d15fc55
    blah_test-83a8b72f-9d4a-45a3-ba39-8a16ff576dea
    blah_test-9ad84ce2-b2cf-4774-9085-1d02256c0860
    blah_test-a9845d07-e661-46e1-8f4a-8b46c9eb7b9d
    blah_test-baa5e9d7-e404-4e85-b2f2-4b4348ba2d1d
    blah_test-cd60d758-c915-4d42-b0ca-24b15fb6ecc7
    blah_test-cf265a0d-060c-4bf5-a80d-188ccc5fd13c
    blah_test-e217d33b-447d-41d5-afda-bd0b44908ca6
    blah_test-ed192204-5b14-4b85-b874-0accf893cee4
    ```
### Expected behavior
Only one set (e.g. number of CPU cores) of parallel test databases are created.

### Actual behavior
A new set of unique databases are created with every test run.

### System configuration
**Rails version**: Edge (459db0a69e1537ac484c26b252ff40543dc0aaed)

**Ruby version**: 2.7.1

**Comment by brenogazzola** [comment]
Having the same problem with a PostgreSQL database.

**Comment by eileencodes** [comment]
Is this behavior the same or different from 6.0?

**Comment by javan** [comment]
It's different. On the mainline gem (6.0.3) test databases have numeric suffixes (`blah_test-0`, `blah_test-1`, etc.).

**Comment by eileencodes** [comment]
Do you know what changed caused this? I didn't make any of the changes to 6.1 parallelization and looking at the code it still uses the same code to create the db configs. 😕 

https://github.com/rails/rails/blob/35bf86fe838d310833adb283712ce4913447eea7/activerecord/lib/active_record/test_databases.rb#L11-L22

**Comment by brenogazzola** [comment]
I think it was https://github.com/rails/rails/commit/b22c4e2a1cc0d8d094e19b5901908ace54cd812e

The `worker` was being initialized with the `id` but removed that parameter and started using `SecureRandom.uuid` internally.

**Comment by eileencodes** [comment]
I pinged @leequarella about this and he's going to take a look today.

**Comment by jhawthorn** [comment]
Thanks so much for the reports 🙇‍♂️. This should be fixed by #39300
